libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -629,33 +679,36 @@ export class PackageExtractor {
     state: IExtractorState
   ): Promise<void> {
     const { includeNpmIgnoreFiles, targetRootFolder } = options;
-    const { projectConfigurationsByPath, archiver } = state;
+    const { projectConfigurationsByPath, packageJsonByPath, dependencyConfigurationsByName, archiver } =
+      state;
     let useNpmIgnoreFilter: boolean = false;
-    let includeFilters: IMinimatch[] | undefined;
-    let excludeFilters: IMinimatch[] | undefined;
 
     const sourceFolderRealPath: string = await FileSystem.getRealPathAsync(sourceFolderPath);
     const sourceProjectConfiguration: IExtractorProjectConfiguration | undefined =
       projectConfigurationsByPath.get(sourceFolderRealPath);
-    if (sourceProjectConfiguration) {
-      if (!includeNpmIgnoreFiles) {
-        // Only use the npmignore filter if the project configuration explicitly asks for it
-        useNpmIgnoreFilter = true;
-      }
-      if (sourceProjectConfiguration.patternsToInclude?.length) {
-        includeFilters = sourceProjectConfiguration.patternsToInclude.map(
-          (p) => new Minimatch(p, { dot: true })
-        );
+
+    const packagesJson: IPackageJson | undefined = packageJsonByPath.get(sourceFolderRealPath);
+    // As this function will be used to copy folder for both project inside monorepo and third party dependencies insides node_modules
+    // Third part dependencies won't have project configurations
+    const isLocalProject: boolean = !!sourceProjectConfiguration;
+
+    // Base filter function
+    const isFileExcluded = (
+      filePath: string,
+      {
+        patternsToInclude,
+        patternsToExclude
+      }: Pick<IExtractorDependencyConfiguration, 'patternsToInclude' | 'patternsToExclude'>
```

Comment:
Since we already know `isLocalProject` and have the package.json, we can just get the correct patterns within this function without needing to pass them in.